### DISCUSSION TITLE
How can I use the PrivateCookieJar in a FromRequest implementation?

### DISCUSSION THREAD

**Discussion by szattila98** [tokio-rs/axum#1055] [Q&A] [answered]
I want to access the private cookies in a FromRequest implementation but I have difficulties getting a PrivateCookieJar there to do so.

**Comment by jplatte** [comment]
"I have difficulties" is not really a useful issue description. Can you please show a snippet of the code you have tried, and explain what behavior you observed (compilation error / runtime error / silent misbehavior)?

**Reply by szattila98** [reply]
Thanks for the fast reply, I was a bit tired yesterday and felt that it was adequate, sorry about it. I solved it but I have a question, I will explain it as a reply to @davidpdrsn

**Comment by davidpdrsn** [comment, marked as answer]
You mean something like this https://docs.rs/axum/latest/axum/extract/index.html#accessing-other-extractors-in-fromrequest-implementations?

**Reply by szattila98** [reply]
Yeah something like that, I solved what I wanted, here it is.
![image](https://user-images.githubusercontent.com/37137309/170533486-a0d46f82-22e3-4e9e-b821-c14e845e5d93.png)
It gets a cookie and creates an AuthenticatedUser so it prevents some boilerplate, especially when I extend the functionality.

But it did bring up a question which is, how does the PrivateCookieJar know what the key is? I have it generated or configured from .env elsewhere, but it knows and parses the private cookies with it.

**Reply by davidpdrsn** [reply]
It requires you to set the key via `Extension` like shown here https://docs.rs/axum-extra/latest/axum_extra/extract/cookie/struct.PrivateCookieJar.html#example and the struct is defined with the key type it uses:

```rust
pub struct PrivateCookieJar<K = Key> { ... }
```

**Reply by szattila98** [reply]
I see, so it can access it internally, thank you very much.
